client/rpc.go

```diff
@@ -110,7 +110,7 @@ func canRetry(args interface{}, err error) bool {
 	// Reads are safe to retry for stream errors, such as if a server was
 	// being shut down.
 	info, ok := args.(structs.RPCInfo)
-	if ok && info.IsRead() && lib.IsErrEOF(err) {
+	if ok && info.IsRead() && err == io.EOF {
```

Comment:
`IsErrEOF`  function checks for other error types, e.g. Yamux closure: https://github.com/hashicorp/consul/blob/v1.7.2/lib/eof.go .  Worth adding it to avoid accidental reversion.

Reply:
Indeed.